compiler/rustc_codegen_llvm/src/asm.rs

```diff
@@ -244,6 +249,30 @@ impl AsmBuilderMethods<'tcx> for Builder<'a, 'll, 'tcx> {
             }
         }
 
+        // Labels should be made local to prevent issues with inlined `asm!` and duplicate labels
+        // Fixes https://github.com/rust-lang/rust/issues/74262
+        for label in labels.iter() {
+            let ranges: Vec<_> = template_str
+                .match_indices(label)
```

Comment:
I don't think this checks if the character *before* the match is a non-identifier character. This means that if you have a label `bar` then `foobar` will be replaced with `foo${:private}bar${:uid}` which is really not what we want.